### PR TITLE
Refactor variable rename to handle UI actions more directly.

### LINKED ISSUE #723: No callback when cancelling variable rename.
Create any variable block.
Choose rename...
Cancel dialog.
Error appears on console.

variables.js:281 Uncaught TypeError: callback is not a function(…)
(anonymous function)    @   variables.js:281
Blockly.prompt  @   blockly.js:377
Blockly.Variables.promptName    @   variables.js:270
Blockly.FieldVariable.classValidator    @   field_variable.js:143
Blockly.Field.callValidator @   field.js:253
callback    @   field_dropdown.js:113

### PR DESCRIPTION
**Pull request by AnmAtAnm**
FieldDropdown now has onItemSelected(..) method to handle the menu item selection action. Variable renames and deletes are now handled here, instead of during validation.
Fixes #723.

---
This change is [Reviewable](https://reviewable.io/reviews/google/blockly/726)

### CODE REVIEW

File: core/field_dropdown.js [hunk 1]
```diff
@@ -193,6 +187,22 @@ Blockly.FieldDropdown.prototype.showEditor_ = function() {
 };
 
 /**
+ * Handle the selection of an item in the dropdown menu.
+ * @param {goog.ui.Menu} menu
```
Comment (rachel-fenichel) [inline core/field_dropdown.js]: param tag should include a description of the parameter, here and below.

Comment (AnmAtAnm) [inline core/field_dropdown.js]: Added descriptions.

File: core/field_dropdown.js [hunk 2]
```diff
@@ -193,6 +187,22 @@ Blockly.FieldDropdown.prototype.showEditor_ = function() {
...
+ * @param {goog.ui.MenuItem} menuItem
+ */
+Blockly.FieldDropdown.prototype.onItemSelected = function(menu, menuItem) {
+  var value = menuItem.getValue();
+  if (thisField.sourceBlock_) {
```
Comment (AnmAtAnm) [inline core/field_dropdown.js]: Fixed.

File: core/workspace.js [hunk 3]
```diff
@@ -301,36 +301,42 @@ Blockly.Workspace.prototype.getVariableUses = function(name) {
...
+    function doDeletion() {
+      Blockly.Events.setGroup(true);
+      for (var i = 0; i < uses.length; i++) {
+        uses[i].dispose(true, false);
+      }
+      Blockly.Events.setGroup(false);
+      workspace.variableList.splice(variableIndex, 1);
+    }
     if (uses.length > 1) {
+      // Confirm the deletion of multiple blocks
```
Comment (rachel-fenichel) [inline core/workspace.js]: No, that's the confirm call on line (new) 329.  This is // Don't delete the variable if it's used in the definition of a procedure.

**Comment by rachel-fenichel** [comment]
One comment fix, then lgtm.

**Pull request merged**